spring-xd-messagebus-rabbit/src/main/java/org/springframework/xd/dirt/integration/rabbit/RabbitMessageBus.java

```diff
@@ -874,4 +878,30 @@ public boolean getAutoBindDLQ(boolean defaultValue) {
 
 	}
 
+	@Override
+	public void storeForManualAck(MessageHeaders headers) {
+		if (headers.containsKey("amqp_channel")) {
+			Channel channel = (com.rabbitmq.client.Channel) headers.get("amqp_channel");
+			if (!channelsForManualAck.containsKey(channel)) {
+				channelsForManualAck.put(channel, 1);
+			}
+			else {
+				channelsForManualAck.put(channel, channelsForManualAck.get(channel) + 1);
+			}
+		}
+	}
+
+	@Override
+	public void doManualAck() {
+		for (final com.rabbitmq.client.Channel channel : channelsForManualAck.keySet()) {
```

Comment:
It's more efficient to iterate over the `entrySet()` to save the hit of `get()`.